### PR TITLE
Agents improvements and dependencies

### PR DESCRIPTION
**Pull request by gdiazlo**
- Make some changes to agents to improve their resilience in some areas. 
- Remove unwanted dependencies.
- Increase memberlist version.
- Include a new qedmanager script to manage all services (wip) in development environment

### CODE REVIEW

File: gossip/monitor/monitor.go [hunk 1]
```diff
@@ -194,21 +203,18 @@ func (m Monitor) sendAlert(msg string) {
...
-	log.Debug("Executing task: %+v", task)
-	resp, err := m.client.Incremental(task.Start, task.End)
+func (q QueryTask) Do() {
+	log.Debug("Executing task: %+v", q)
+	resp, err := q.client.Incremental(q.Start, q.End)
 	if err != nil {
-		// TODO: retry
 		metrics.QedMonitorGetIncrementalProofErrTotal.Inc()
-		log.Infof("Unable to get incremental proof from QED server: %s", err.Error())
+		log.Infof("Unable to verify incremental proof from %d to %d", q.Start, q.End)
```
Comment (jllucas) [inline gossip/monitor/monitor.go]: This error message doesn't tell the true error source. The former was correct.

File: go.mod [hunk 2]
```diff
@@ -1,45 +1,21 @@
 module github.com/bbva/qed
 
+go 1.12
```
Comment (jllucas) [inline go.mod]: go 1.12 is required?

File: gossip/agent_test.go [hunk 3]
```diff
@@ -69,7 +69,7 @@ func TestLeave(t *testing.T) {
 	conf := DefaultConfig()
 	conf.NodeName = "testNode"
 	conf.Role = member.Auditor
-	conf.BindAddr = "0.0.0.0:12346"
+	conf.BindAddr = "127.0.0.1:12346"
```
Comment (jllucas) [inline gossip/agent_test.go]: I would set default bind addr to empty string

File: gossip/agent_test.go [hunk 4]
```diff
@@ -117,7 +117,7 @@ func TestShutdown(t *testing.T) {
 	conf := DefaultConfig()
 	conf.NodeName = "testNode"
 	conf.Role = member.Auditor
-	conf.BindAddr = "0.0.0.0:12347"
+	conf.BindAddr = "127.0.0.1:12347"
```
Comment (jllucas) [inline gossip/agent_test.go]: I would set default bind addr to empty string

File: gossip/config.go [hunk 5]
```diff
@@ -29,7 +29,7 @@ const DefaultBindPort int = 7946
 // DefaultConfig contains the defaults for configurations.
 func DefaultConfig() *Config {
 	return &Config{
-		BindAddr:            "0.0.0.0:12345",
+		BindAddr:            "127.0.0.1:12345",
```
Comment (jllucas) [inline gossip/config.go]: I would set default bind addr to empty string

File: gossip/agent.go [hunk 6]
```diff
@@ -106,6 +106,17 @@ func NewAgent(conf *Config, p []Processor) (agent *Agent, err error) {
 
 	return agent, nil
 }
+func chTimedSend(batch *protocol.BatchSnapshots, ch chan *protocol.BatchSnapshots) {
+	for {
+		select {
+		case <-time.After(200 * time.Millisecond):
+			log.Infof("Timed out sending out batch ")
```
Comment (jllucas) [inline gossip/agent.go]: "Timed out enqueueing batch in out channel"

File: gossip/agent_test.go (hunk 4 above)
Comment (gdiazlo) [reply gossip/agent_test.go]: The tests starts an agent, so it needs a bindAddr. It is in fact changing the DefaultConfig() to have a valid bindaddr. I think we can leave this as it is for now.

File: gossip/config.go (hunk 5 above)
Comment (gdiazlo) [reply gossip/config.go]: I'll change this.

File: gossip/agent_test.go (hunk 3 above)
Comment (gdiazlo) [reply gossip/agent_test.go]: The tests starts an agent, so it needs a bindAddr. It is in fact changing the DefaultConfig() to have a valid bindaddr. I think we can leave this as it is for now.

File: go.mod (hunk 2 above)
Comment (gdiazlo) [reply go.mod]: Go 1.11.4 seems to work too with that line. I suppose the modules thing is going to change these days in preparation to the final release in go 1.13 this summer.

File: gossip/monitor/monitor.go (hunk 1 above)
Comment (gdiazlo) [reply gossip/monitor/monitor.go]: Fixed.

File: gossip/agent.go (hunk 6 above)
Comment (gdiazlo) [reply gossip/agent.go]: Fixed

**Pull request merged**